### PR TITLE
feat: Add WynntilsStatisticsScreen

### PR DESCRIPTION
**Pull request by kristofbolyai**
Depends on #1852

image
image
image
image

### CODE REVIEW

**Comment by magicus** [comment]
Some comments from just looking at the screenshots:

1) Maybe add the total to the tooltip? That is kind of a compromise of what I wanted and what is possible to do with the format. And you still would not have to click on every statistic to see the value.

2) We have `StringUtils.formatAmount()` which would make like `6.1m` of `6129321`. Maybe use that instead, and only display the complete number if you press down shift, or select a toggle.

**Comment by kristofbolyai** [comment]
> We have StringUtils.formatAmount() which would make like 6.1m of 6129321. Maybe use that instead, and only display the complete number if you press down shift, or select a toggle.

In the tooltip?

**Comment by magicus** [comment]
> In the tooltip?

No, not only, also at the side page. At some point it is just hard for humans to read and understand large numbers. Compare:

Count: 78.2 k

with

Count: 78177

**Comment by kristofbolyai** [comment]
> > In the tooltip?
> 
> 
> 
> No, not only, also at the side page. At some point it is just hard for humans to read and understand large numbers. Compare:
> 
> 
> 
> Count: 78.2 k
> 
> 
> 
> with
> 
> 
> 
> Count: 78177
> 
> 

But the point of statistics is to get accurate info?

**Comment by kristofbolyai** [comment]
image

File: common/src/main/java/com/wynntils/services/statistics/CustomStatFormatters.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.statistics;
+
+import net.minecraft.stats.StatFormatter;
+
+public final class CustomStatFormatters {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/services/statistics/CustomStatFormatters.java:9]:
```suggestion
public final class CustomStatFormatters implements StatFormatter {
```

?

Comment (magicus) [reply common/src/main/java/com/wynntils/services/statistics/CustomStatFormatters.java:9]: No, sorry, that's not how any of this work! 🤦

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/services/statistics/CustomStatFormatters.java:9]: I am not sure how this helps? We don't create our own interface/abstract class, we just needed a class to hold our custom formatters.

**Comment by kristofbolyai** [comment]
I did not implement `StringUtils.formatAmount()` because it needs a new formatter to work properly. We should also make a way to hide certain stat fields where it doesnt matter (like spells cast). That needs to be a follow up.

**Comment by magicus** [comment]
> But the point of statistics is to get accurate info?

Yeah, but if you have a value in the millions, small differences is not relevant. I mean, you display the total lootrun time as "3.25 h", even if you could have said 12300 seconds. Or maybe 12344 seconds; we don't know since you lose accuracy. But that would not be very helpful.

Once again, I am all in favor of making it possible to see the exact details (and in that case, maybe also seeing every second for time based entries), but this should not be the default, but only if you e.g. hold down shift. Or, we should have a simple toggle button for "compact" or "full" representation.

Basically, I am arguing that by default we should pass the value through `StringUtils.integerToShortString()`.

**Comment by kristofbolyai** [comment]
Is it acceptable like this, and I will work on filtering/formatting improvements today, separately?

**Comment by magicus** [comment]
Okay.

**Review by magicus — APPROVED** [review]
